File: Livro - Módulo Java/5 - Múltipla Escolha/Exercicio.06.java
//   6 - Repita o exercício anterior considerando que os menus sejam de uma lanchonete, e são os seguintes: 
// - As bebidas e as quantidades vendidas 
// - Os lanches e as quantidades vendidas 
// - As porções e as quantidades consumidas 
// No final do processa // a. A maior e a menor conta 
// b. A média de gasto por mesa 
// c. A média de gasto por pessoa  . A média de lanches consumidos por mesa 
// e. A média de lanches consumidos por pessoas 
// f. A média de porções consumidas por mesa 

import java.util.Scanner;

public class Lanchonete {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int numMesas;
        double totalConta, maiorConta, menorConta, mediaGastoMesa, mediaGastoPessoa,
                mediaLanchesPorMesa, mediaLanchesPorPessoa, mediaPorcoesPorMesa, totalLanches, totalPorcoes;

        System.out.print("Informe o número de mesas na lanchonete: ");
        numMesas = scanner.nextInt();

        maiorConta = menorConta = totalConta = mediaGastoMesa = mediaGastoPessoa = 0;
        mediaLanchesPorMesa = mediaLanchesPorPessoa = mediaPorcoesPorMesa = totalLanches = totalPorcoes = 0;

        for (int mesa = 1; mesa <= numMesas; mesa++) {
            System.out.println("Mesa " + mesa + ":");

            // Bebidas
            System.out.print("Quantidade de bebidas vendidas: ");
            int bebidas = scanner.nextInt();

            // Lanches
            System.out.print("Quantidade de lanches vendidos: ");
            int lanches = scanner.nextInt();
            mediaLanchesPorMesa += lanches;

            // Porções
            System.out.print("Quantidade de porções consumidas: ");
            int porcoes = scanner.nextInt();
            mediaPorcoesPorMesa += porcoes;

            // Calcular conta da mesa
            double contaMesa = calcularConta(bebidas, lanches, porcoes);
            totalConta += contaMesa;

            // Atualizar maior e menor conta
            if (mesa == 1 || contaMesa > maiorConta) {
                maiorConta = contaMesa;
            }
            if (mesa == 1 || contaMesa < menorConta) {
                menorConta = contaMesa;
            }

            // Atualizar total de lanches e porções
            totalLanches += lanches;
            totalPorcoes += porcoes;
        }

        // Calcular médias
        mediaGastoMesa = totalConta / numMesas;
        mediaGastoPessoa = totalConta / (numMesas * 4); // Assumindo 4 pessoas por mesa
        mediaLanchesPorMesa /= numMesas;
        mediaLanchesPorPessoa = mediaLanchesPorMesa / (numMesas * 4);
        mediaPorcoesPorMesa /= numMesas;

        // Imprimir resultados
        System.out.println("\nResultados:");
        System.out.println("a. Maior conta: " + maiorConta);
        System.out.println("   Menor conta: " + menorConta);
        System.out.println("b. Média de gasto por mesa: " + mediaGastoMesa);
        System.out.println("c. Média de gasto por pessoa: " + mediaGastoPessoa);
        System.out.println("d. Média de lanches consumidos por mesa: " + mediaLanchesPorMesa);
        System.out.println("e. Média de lanches consumidos por pessoa: " + mediaLanchesPorPessoa);
        System.out.println("f. Média de porções consumidas por mesa: " + mediaPorcoesPorMesa);
        System.out.println("g. Total de lanches: " + totalLanches);
        System.out.println("   Total de porções: " + totalPorcoes);

        scanner.close();
    }

    // Função para calcular a conta da mesa
    private static double calcularConta(int bebidas, int lanches, int porcoes) {
        double precoBebida = 3.0; // Preço por bebida
        double precoLanche = 15.0; // Preço por lanche
        double precoPorcao = 10.0; // Preço por porção

        return (bebidas * precoBebida) + (lanches * precoLanche) + (porcoes * precoPorcao);
    }
}

File: Livro - Módulo Python/5 - Múltipla Escolha/Exercicio.06.py
#6. Faça um algoritmo para calcular a conta de um número indefinidos de mesas de uma lanchonete.
# Considere que o usuário possa informar através de um menu os seguintes itens:
# - As bebidas e as quantidades vendidas
# - Os lanches e as quantidades vendidas
# - As porções e as quantidades consumidas
# No final do processamento (no final do cálculo de todas as contas das mesas) as seguintes informações devem ser impressas:
# a. A maior e a menor conta
# b. A média de gasto por mesa
# c. A média de gasto por pessoa
# d. A média de lanches consumidos por mesa
# e. A média de lanches consumidos por pessoas
# f. A média de porções consumidas por mesa
# g. O total de lanche e de porções

maiorConta = 0.0
menorConta = float('inf')
somaContas = 0.0
somaPessoas = 0
somaLanches = 0
somaPorcoes = 0
totalLanches = 0
totalPorcoes = 0
quantidadeMesas = 0

while True:
    opcao = input("Deseja calcular a conta de uma mesa? (S/N): ")

    if opcao.upper() == 'N':
        break

    qtdeBebidas = int(input("Digite a quantidade de bebidas vendidas: "))
    qtdeLanches = int(input("Digite a quantidade de lanches vendidos: "))
    qtdePorcoes = int(input("Digite a quantidade de porções consumidas: "))

    qtdePessoas = int(input("Digite o número de pessoas na mesa: "))

    valorBebidas = qtdeBebidas * 2.50
    valorLanches = qtdeLanches * 10.00
    valorPorcoes = qtdePorcoes * 15.00
    valorConta = valorBebidas + valorLanches + valorPorcoes

    maiorConta = max(maiorConta, valorConta)
    menorConta = min(menorConta, valorConta)
    somaContas += valorConta
    somaPessoas += qtdePessoas
    somaLanches += qtdeLanches
    somaPorcoes += qtdePorcoes
    totalLanches += qtdeLanches
    totalPorcoes += qtdePorcoes

    quantidadeMesas += 1

if quantidadeMesas > 0:
    mediaGastoMesa = somaContas / quantidadeMesas
    mediaGastoPessoa = somaContas / somaPessoas
    mediaLanchesMesa = somaLanches / quantidadeMesas
    mediaLanchesPessoa = somaLanches / somaPessoas
    mediaPorcoesMesa = somaPorcoes / quantidadeMesas
else:
    mediaGastoMesa = 0
    mediaGastoPessoa = 0
    mediaLanchesMesa = 0
    mediaLanchesPessoa = 0
    mediaPorcoesMesa = 0

print("\nRESUMO DAS CONTAS DAS MESAS")
print(f"Maior conta: R$ {maiorConta:.2f}")
print(f"Menor conta: R$ {menorConta:.2f}")
print(f"Média de gasto por mesa: R$ {mediaGastoMesa:.2f}")
print(f"Média de gasto por pessoa: R$ {mediaGastoPessoa:.2f}")
print(f"Média de lanches consumidos por mesa: {mediaLanchesMesa:.2f}")
print(f"Média de lanches consumidos por pessoa: {mediaLanchesPessoa:.2f}")
print(f"Média de porções consumidas por mesa: {mediaPorcoesMesa:.2f}")
print(f"Total de lanches vendidos: {totalLanches}")
print(f"Total de porções consumidas: {totalPorcoes}")